egs/sprakbanken_swe/s5/local/sprak_data_prep.sh

```diff
@@ -0,0 +1,136 @@
+traindir=`pwd`/data/local/trainsrc
+testdir=`pwd`/data/local/testsrc
+rm -rf $lmdir $traindir $testdir $devdir
+mkdir -p $dir $lmdir $traindir $testdir $devdir
+local=`pwd`/local
+utils=`pwd`/utils
+
+
+. ./path.sh
+
+# Checks if python3 is available on the system and install python3 in userspace if not
+# This recipe currently relies on version 3 because python3 uses utf8 as internal 
+# string representation
+
+if ! which python3 >&/dev/null; then
+  echo "Installing python3 since not on your path."
+  pushd $KALDI_ROOT/tools || exit 1;
+  extras/install_python3.sh || exit 1;
+  popd
+fi
+
+if [ ! -d $dir/download ]; then
+    mkdir -p $dir/download/0467-1 $dir/download/0467-2 $dir/download/0467-3
+fi 
+
+echo "Downloading and unpacking sprakbanken to $dir/corpus_processed. This will take a while."
+
+if [ ! -f $dir/download/sve.16khz.0467-1.tar.gz ]; then 
+    ( wget http://www.nb.no/sbfil/talegjenkjenning/16kHz/sve.16khz.0467-1.tar.gz --directory-prefix=$dir/download ) &
+fi
+
+if [ ! -f $dir/download/sve.16khz.0467-2.tar.gz ]; then 
+    ( wget http://www.nb.no/sbfil/talegjenkjenning/16kHz/sve.16khz.0467-2.tar.gz --directory-prefix=$dir/download ) &
+fi
+
+if [ ! -f $dir/download/sve.16khz.0467-3.tar.gz ]; then 
+    ( wget http://www.nb.no/sbfil/talegjenkjenning/16kHz/sve.16khz.0467-3.tar.gz --directory-prefix=$dir/download ) &
+fi
+
+if [ ! -f $dir/download/sve.16khz.0467-1.tar.gz ]; then 
+    ( wget http://www.nb.no/sbfil/talegjenkjenning/16kHz/sve.16khz.0468.tar.gz --directory-prefix=$dir/download ) &
+fi    
+wait
+
+echo "Corpus files downloaded."
+
+if [ ! -d $dir/download/0468 ]; then
+    echo "Unpacking files."
+    tar -xzf $dir/download/sve.16khz.0467-1.tar.gz -C $dir/download/0467-1
+    tar -xzf $dir/download/sve.16khz.0467-2.tar.gz -C $dir/download/0467-2 &
+    tar -xzf $dir/download/sve.16khz.0467-3.tar.gz -C $dir/download/0467-3 &
+    tar -xzf $dir/download/sve.16khz.0468.tar.gz -C $dir/download/0468    
+
+    wait     
+    echo "Corpus unpacked succesfully."
+fi
+
+
+sph2pipe=$KALDI_ROOT/tools/sph2pipe_v2.5/sph2pipe
+if [ ! -x $sph2pipe ]; then
+   echo "Could not find (or execute) the sph2pipe program at $sph2pipe";
+   exit 1;
+fi
+wait
+echo "done"
+
+echo "Converting downloaded files to a format consumable by Kaldi scripts."
+
+rm -rf $dir/corpus_processed 
+mkdir -p $dir/corpus_processed/training/0467-1 $dir/corpus_processed/training/0467-2 $dir/corpus_processed/training/0467-3 
+
+# Create parallel file lists and text files, but keep sound files in the same location to save disk space
+# Writes the lists to data/local/data (~ 310h)
+echo "Creating parallel data for training data."
+python3 $local/sprak2kaldi.py $dir/download/0467-1 $dir/corpus_processed/training/0467-1 &  # ~140h
+python3 $local/sprak2kaldi.py $dir/download/0467-2 $dir/corpus_processed/training/0467-2 &  # ~125h
+python3 $local/sprak2kaldi.py $dir/download/0467-3 $dir/corpus_processed/training/0467-3 &  # ~128h
+
```

Comment:
Again, I don't think running these things in parallel is a good general strategy. Usually it only makes things much more complicated to debug and the speed increase is marginal (especially in multi-user NFS environment)